# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/opengl/TextureClientOGL.cpp
# Commit: 27fa39e98e4c
# Full Hash: 27fa39e98e4cffdbf67935cedc6276771f0b330d
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-01-25 16:29:55
# Description:
#   Bug 1811929 - Fix JNI crash caused by passing incorrect jobject type. r=gfx-reviewers,nical
#   
#   The function ANativeWindow_fromSurface() takes a jobject that is
#   supposed to be a Surface. Prior to bug 1706656 GeckoSurface was a
#   subclass of Surface, meaning we passed the correct type. However,
# ==============================================================================

diff -r 5a5d0c9e338e -r 27fa39e98e4c gfx/layers/opengl/TextureClientOGL.cpp
--- a/gfx/layers/opengl/TextureClientOGL.cpp	Tue Jan 24 15:31:03 2023 +0000
+++ b/gfx/layers/opengl/TextureClientOGL.cpp	Tue Jan 24 15:31:31 2023 +0000
@@ -111,8 +111,8 @@
     java::GeckoSurface::Param aSurface, gfx::IntSize aSize,
     gfx::SurfaceFormat aFormat)
     : mSurface(aSurface), mIsLocked(false), mSize(aSize), mFormat(aFormat) {
-  mNativeWindow =
-      ANativeWindow_fromSurface(jni::GetEnvForThread(), mSurface.Get());
+  mNativeWindow = ANativeWindow_fromSurface(jni::GetEnvForThread(),
+                                            mSurface->GetSurface().Get());
   MOZ_ASSERT(mNativeWindow, "Failed to create NativeWindow.");
 
   // SurfaceTextures don't technically support BGR, but we can just pretend to
